File: backend/app/schemas/file.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class FileMetadataResponse(BaseModel):
    """Response schema for a single file's metadata (matches DB model)."""
    model_config = ConfigDict(from_attributes=True)

    id: int
    filename: str
    size: int | None
    rows: int
    columns: int
    uploaded_at: datetime


class ListFilesResponse(BaseModel):
    """Response schema for paginated file list."""
    data: list[FileMetadataResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class UploadResponse(BaseModel):
    """Response schema after successful upload."""
    filename: str
    rows: int
    cols: int
    size: int

File: frontend/src/types/file.ts
export interface FileMetadata {
  id: number;
  filename: string;
  file_path?: string;
  rows: number;
  columns: number;
  uploaded_at: string;
  size?: number;
}

export interface GetFilesResponse {
  data: FileMetadata[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface GetFileDataResponse {
  data: Record<string, string>[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

  